odk/src/main/java/org/motechproject/odk/service/impl/FormDefinitionImportServiceKobo.java

```diff
@@ -0,0 +1,148 @@
+            formElements.add(new FormElementBuilder().setName(KoboConstants.BAMBOO_DATASET_ID).setLabel(KoboConstants.BAMBOO_DATASET_ID).setType(FieldTypeConstants.STRING).createFormElement());
+            formElements.add(new FormElementBuilder().setName(OnaConstants.TAGS).setLabel(OnaConstants.TAGS).setType(FieldTypeConstants.STRING_ARRAY).createFormElement());
+            formElements.add(new FormElementBuilder().setName(OnaConstants.SUBMITTED_BY).setLabel(OnaConstants.SUBMITTED_BY).setType(FieldTypeConstants.STRING).createFormElement());
+            formElements.add(new FormElementBuilder().setName(OnaConstants.XFORM_ID_STRING).setLabel(OnaConstants.XFORM_ID_STRING).setType(FieldTypeConstants.STRING).createFormElement());
+            formElements.add(new FormElementBuilder().setName(OnaConstants.SUBMISSION_TIME).setLabel(OnaConstants.SUBMISSION_TIME).setType(FieldTypeConstants.DATE_TIME).createFormElement());
+            formElements.add(new FormElementBuilder().setName(OnaConstants.GEOLOCATION).setLabel(OnaConstants.GEOLOCATION).setType(FieldTypeConstants.DOUBLE_ARRAY).createFormElement());
+            formElements.add(new FormElementBuilder().setName(KoboConstants.USERFORM_ID).setLabel(KoboConstants.USERFORM_ID).setType(FieldTypeConstants.STRING).createFormElement());
+            formElements.add(new FormElementBuilder().setName(OnaConstants.STATUS).setLabel(OnaConstants.STATUS).setType(FieldTypeConstants.STRING).createFormElement());
+            formElements.add(new FormElementBuilder().setName(OnaConstants.ID).setLabel(OnaConstants.ID).setType(FieldTypeConstants.INT).createFormElement());
+        }
+    }
+
+    private void alterFormElementName(FormElement formElement) {
+        String formFieldName = formElement.getName();
+        String name = formFieldName.substring(formFieldName.indexOf('/', 1) + 1, formFieldName.length());
+        formElement.setName(name);
+
+        if (formElement.hasChildren()) {
+            for (FormElement child : formElement.getChildren()) {
+                alterFormElementName(child);
+            }
+        }
+    }
+
+    @Override
+    protected List<String> parseToUrlList(String responseBody) throws ParseUrlException {
+        try {
+            List<KoboFormInfo> koboFormInfoList = objectMapper.readValue(responseBody, type);
+            List<String> urls = new ArrayList<>();
+
+            for (KoboFormInfo koboFormInfo : koboFormInfoList) {
+                urls.add(koboFormInfo.getUrl());
+            }
+            return urls;
+        } catch (IOException e) {
+            throw new ParseUrlException(e);
+        }
+
+    }
+
+    private String buildFormsQuery(Configuration configuration) {
+        return configuration.getUrl() + API_PATH + FORMS_PATH + OWNER_QUERY + configuration.getUsername();
+    }
+
+    private String normalizeFormDef(String formDef) {
+        if (formDef.startsWith("<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
```

Comment:
well this looks really suspect. Why are we doint this? A comment at least would be helpful